shumai/tensor/register_gradients.ts

```diff
@@ -10,6 +10,18 @@ export interface Grad {
   out: Tensor
 }
 
+function recoverShape(tensor: Tensor, originalShape: number[], lostAxes: number[]) {
+  const shapeForBroadcast = [...originalShape]
+  for (let axis of lostAxes) {
+    if (axis < 0) {
+      axis += originalShape.length
+    }
+    shapeForBroadcast[axis] = 1
+  }
+  const tensorForBroadcast = tensor.reshape(shapeForBroadcast)
+  return tensorForBroadcast.add(sm.full(originalShape, 0))
```

Comment:
@jacobkahn @StrongerXi, do y'all know if this will likely be JIT compiled away? (addition of 0 to induce broadcast)

Reply:
It'll be pretty easy to pick up and rewrite such pattern in the JIT I'm working on.

Could you give an example of this add op (with input shapes)?

If I understand this correctly, the goal is to potentially expand `tensorForBroadcast` in some axes, which sounds like `fl::tile`?

```
Tensor t = rand(Shape(1, 3, 1));

t + full(0, Shape(3, 3, 3))
===>
fl::tile(t, Shape(3, 1, 3))
```

Reply:
Hi, yes that's what I was trying to do. I was trying to skip the extra logic for determining the tile shape.

Reply:
@jacobkahn - this seems like an FL bug?

@yushiyangk - I'd avoid the `deepCopy()` for now and leave the impl as is :)

Reply:
Got it, I've updated the PR with the original version of `recoverShape`, thanks!